clang/lib/Index/USRGeneration.cpp

```diff
@@ -1070,6 +1063,24 @@ void USRGenerator::VisitMSGuidDecl(const MSGuidDecl *D) {
   D->NamedDecl::printName(Out);
 }
 
+void USRGenerator::VisitBaseUsingDecl(const BaseUsingDecl *D) {
+  // Add the filename when needed to disambiguate using decls from different
+  // files.
+  if (ShouldGenerateLocation(D) && GenLoc(D, /*IncludeOffset=*/false))
```

Comment:
as discussed offline, ShouldGenerateLocation doesn't seem like the right check here:
 - we definitely want to include the filename for *.cpp
 - we may want to include the filename for *.h
 - we probably should exclude system headers
 - for using decls inside functions, we need to include the filename if the **function** is not externally visible (though it's probably fine to include it always)